dockstore-webservice/src/main/java/io/openapi/api/impl/ToolsApiServiceImpl.java

```diff
@@ -288,24 +289,41 @@ public Response toolsGet(String id, String alias, String toolClass, String descr
         } else if (alias != null) {
             all.add(toolDAO.getGenericEntryByAlias(alias));
         } else {
-            if (toolClass == null || COMMAND_LINE_TOOL.equalsIgnoreCase(toolClass)) {
+            // If there is no matching descriptor language, then do not add any tools or workflows.
+            DescriptorLanguage descriptorLanguage = null;
+            boolean descriptorLanguageFound = true;
+
+            // Tricky case for GALAXY because it doesn't match the rules of the other languages
+            if ("galaxy".equalsIgnoreCase(descriptorType)) {
+                descriptorType = DescriptorLanguage.GXFORMAT2.getShortName();
+            }
+
+            if (descriptorType != null) {
+                try {
+                    descriptorLanguage = DescriptorLanguage.convertShortStringToEnum(descriptorType);
+                } catch (UnsupportedOperationException ex) {
+                    LOG.info(ex.getMessage());
+                    descriptorLanguageFound = false;
+                }
+            }
+
+            // TODO: Have DescriptorLanguage indicate whether the language supports tools. Make this less hack-ish
+            // Only WDL and CWL tools are supported on Dockstore.
+            if (descriptorLanguageFound && (toolClass == null || COMMAND_LINE_TOOL.equalsIgnoreCase(toolClass))
```

Comment:
This is nit-picking, but this is a very long condition and hard to understand

1. You already know if the descriptorLanguage has been found by virtue of it being null or not. I don't think you need the `descriptorLanguageFound` variable, although you can argue it enhances readability.
2. With enums, you can use `==`

I would suggest two nested if statements for readability. But that's my bias.

Reply:
Yeah the name and how it's being used doesn't match completely.

The user provides the descriptorType which could be null,  a string that matches one of the languages we support, or a string that doesn't match. I set descriptorLanguageFound to be true unless we are unable to match a string with the language we support. So, if they didn't provide one (null) or it matches, it will stay true and the tools/workflows will be added.

I will try to improve the readability since it is confusing.

Reply:
Ah, I see, so I was wrong about line 316 -- it would still execute...

I would suggest if there is a an invalid descriptor language, just short-circuit and `return Collections.emptyList()`. Then it simplifies your logic downstream.